source/simulator/helper_functions.cc

```diff
@@ -1728,6 +1728,107 @@ namespace aspect
   }
 
 
+  template <int dim>
+  void Simulator<dim>::interpolate_material_output_into_field ()
```

Comment:
I think the problem is that this function goes through every compositional field and interpolates the copy outputs onto all of these fields. I think what you really want to do here is instead of looping over all the fields (below), you want to pass this function the compositional field that you want to interpolate your outputs onto as an argument. Then, in solver_schemes.cc, you could call `interpolate_material_output_into_field(c)`, and then is this function the interpolation would only be done for that specific field.